### ISSUE TITLE
Can't build with `make build-images`

### ISSUE THREAD

**Issue by robertomlsoares** [GoogleContainerTools/kpt-config-sync#245]
Hi! I'm trying to build Config Sync for the first time locally, but it's failing when using `make build-images`. I'm attaching some logs below. Any advice on what could be causing this issue?

I can reproduce with the following steps:
1. `git clone git@github.com:GoogleContainerTools/kpt-config-sync.git`
2. `cd kpt-config-sync`
3. `make build-images`

However, this is not reproducible for some colleagues that have managed to build this repo a few weeks ago. I have tried to build it for the first time a couple of days ago, so perhaps there were some dependency changes recently that could be causing this?

```bash
❯ make build-images

go mod tidy -compat=1.17
Adding credentials for all GCR repositories.
WARNING: A long list of credential helpers may cause delays running 'docker build'. We recommend passing the registry name to configure only the registry you are using.
gcloud credential helpers already registered correctly.
+++ Building the nomos image: gcr.io/[project]/[user]/nomos:v1.13.0-118-gf5ef8e6
[+] Building 0.0s (19/23)
 => [internal] load build definition from Dockerfile                                                                                          0.0s
 => => transferring dockerfile: 37B                                                                                                           0.0s
 => [internal] load .dockerignore                                                                                                             0.0s
 => => transferring context: 34B                                                                                                              0.0s
 => [internal] load metadata for k8s.gcr.io/build-image/debian-base-amd64:bullseye-v1.4.1                                                     0.4s
 => [internal] load metadata for docker.io/library/golang:1.17                                                                                1.7s
 => [internal] load build context                                                                                                             0.5s
 => => transferring context: 668.44kB                                                                                                         0.5s
 => [bins 1/7] FROM docker.io/library/golang:1.17@sha256:87262e4a4c7db56158a80a18fefdc4fee5accc41b59cde821e691d05541bbb18                     0.0s
 => CACHED [nomos  1/11] FROM k8s.gcr.io/build-image/debian-base-amd64:bullseye-v1.4.1@sha256:a74834abd9ef451a735aab8740a2bbac335a14e9c90761  0.0s
 => CANCELED [nomos  2/11] RUN apt-get update &&   apt-get -y upgrade &&   apt-get install -y bash git                                        0.5s
 => CACHED [bins 2/7] WORKDIR /workspace                                                                                                      0.0s
 => CACHED [bins 3/7] COPY . .                                                                                                                0.0s
 => CACHED [bins 4/7] RUN wget https://get.helm.sh/helm-v3.9.0-linux-amd64.tar.gz -O /tmp/helm-v3.9.0-linux-amd64.tar.gz &&   tar -zxvf /tmp  0.0s
 => CACHED [bins 5/7] RUN wget https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize%2Fv4.5.2/kustomize_v4.5.2_linux_amd6  0.0s
 => CACHED [bins 6/7] RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on   go install github.com/GoogleContainerTools/kpt-functions-ca  0.0s
 => CACHED [bins 7/7] RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on   go install     -mod=vendor     -ldflags "-X kpt.dev/configs  0.0s
 => CACHED [nomos  3/11] RUN mkdir -p /opt/nomos/bin                                                                                          0.0s
 => CACHED [nomos  4/11] WORKDIR /opt/nomos/bin                                                                                               0.0s
 => ERROR [nomos  5/11] COPY --from=bins /go/bin/git-importer git-importer                                                                    0.0s
 => ERROR [nomos  6/11] COPY --from=bins /go/bin/monitor monitor                                                                              0.0s
 => ERROR [nomos  7/11] COPY --from=bins /go/bin/nomos nomos                                                                                  0.0s
------
 > [nomos  5/11] COPY --from=bins /go/bin/git-importer git-importer:
------
------
 > [nomos  6/11] COPY --from=bins /go/bin/monitor monitor:
------
------
 > [nomos  7/11] COPY --from=bins /go/bin/nomos nomos:
------
failed to compute cache key: "/go/bin/nomos" not found: not found
make: *** [build-images-monorepo] Error 1
```

**Comment by sdowell** [comment]
I just verified with a fresh clone that I can build all of the images. I'm not sure exactly what went wrong in your use case but I suspect the cached layers may be the culprit. Could you try forcing docker to rebuild those layers using the `--no-cache` option?

You can try passing it in through the `DOCKER_BUILD_QUIET` env var, e.g.
```shell
DOCKER_BUILD_QUIET="--no-cache --quiet" make build-images
```

If you prefer to see the full docker output you can also omit the `--quiet` flag.

**Comment by robertomlsoares** [comment]
Hi @sdowell! Thanks for looking into it. Unfortunately that flag didn't help. I've since stopped debugging this as I have tried a lot of things and couldn't get it to work. We'll also move this flow to our internal build system to not rely on local dev machines for this, but I still thought it was odd that I couldn't build it locally. Feel free to close the ticket if you want, I just thought it was worth letting you know about it.
